container_api/src/config.rs

```diff
@@ -0,0 +1,223 @@
+use holochain_core_types::{
+    entry::agent::{Agent, Identity},
+    error::{HcResult, HolochainError},
+    json::JsonString,
+};
+use holochain_dna::Dna;
+use serde::Deserialize;
+use std::{convert::TryFrom, fs::File, io::prelude::*};
+
+#[derive(Deserialize)]
+pub struct Configuration {
+    #[serde(rename = "agent")]
+    agents: Option<Vec<AgentConfiguration>>,
+    #[serde(rename = "dna")]
+    dnas: Option<Vec<DNAConfiguration>>,
+    #[serde(rename = "instance")]
+    instances: Option<Vec<InstanceConfiguration>>,
+    #[serde(rename = "interface")]
+    interfaces: Option<Vec<InterfaceConfiguration>>,
+    #[serde(rename = "bridge")]
+    bridges: Option<Vec<Bridge>>,
```

Comment:
Might be more idiomatic to just use the plural names in the toml config too, e.g. `[[bridges]]`. The [toml docs](https://github.com/toml-lang/toml#array-of-tables) suggest this.

Reply:
Had it that way before, @sphinxc0re convinced me to use singular in the files, which feels like more natural read of the config. Don't have a strong opinion but like singular more since we will have several blocks of these and each block is just one instance..

Reply:
I'd prefer to keep it. Also, I haven't used toml much. Does it feel strange to have singular array instances? To me it feels strange to have them be plural.
Will change if you feel strongly it should be plural.

Reply:
I agree that it *looks* cleaner singular, but note that this equivalent TOML is also valid (an array with inline tables):

```
agent = [ 
    {id = "test agent 1", name = "Holo Tester 1", key_file = "holo_tester1.key"},
    {id = "test agent 2", name = "Holo Tester 2", key_file = "holo_tester2.key"}
]
```

in which case it clearly should be plural. I think it's more that TOML syntax is awkward for arrays. So I do feel it should be plural, but I don't feel strongly enough about it to block this PR.

Reply:
Ok, I've switched to plurals.